### PR TITLE
fix for remove_partitions method

### PR DESCRIPTION
**Pull request by karolinavelkaja**
Signed-off-by: Karolina Rogowska <karolina.rogowsky@gmail.com>

### CODE REVIEW

File: test_tools/disk_utils.py [hunk 1]
```diff
@@ -232,6 +232,7 @@ def remove_partitions(device):
         .count(1) \
         .block_size(Size(1, Unit.Blocks4096))\
         .oflag("direct")
+    TestRun.executor.run(f"sleep 1")
```
Comment (katlapinka) [inline test_tools/disk_utils.py:235]: I'm just curious - why is this needed here? Did partitions unmounting process not finish before running dd?

Comment (Deixx) [reply test_tools/disk_utils.py:235]: Short comment could be added (NOT the full explanation below ;))
We've run into a problem on one particular platform and only with HDD disks. Partitions would not disappear after dd was run and it looked like it was a race condition. `ls /dev/sdx -1` had displayed either the device and its partition (in that case partition would stay in the system), or only the device, but with error that partition cannot be accessed. I don't know how exactly ls works, but it looked like some initial scan "saw" the partition, but it was not there for the proper scan.  With trial and error we concluded that `mdadm --zero-superblock` was not releasing the device in time.
Actually maybe the sleep should be moved to the end of Mdadm.zero_superblock()... Removing partitions is quite common operation (opposed to zeroing raid metadata).

**Review by Deixx — CHANGES_REQUESTED** [review]
Move the sleep to Mdadm.zero_superbloct()

File: test_tools/disk_utils.py (hunk 1 above)
Comment (robertbaldyga) [reply test_tools/disk_utils.py:235]: > With trial and error we concluded that mdadm --zero-superblock was not releasing the device in time.

Whoa, it looks like a bug in `mdadm`, doesn't it?

**Comment by karolinavelkaja** [comment]
Closing this PR since better resolution had been found and placed under review in ocl: https://github.com/Open-CAS/open-cas-linux/pull/597
Problem: mdadm --zero-superblock command triggers udev and before calling next command it's needed to wait until this event ends. 
Resolution: Instead of using sleep command it's better/safer to call "udevadm settle" which would wait for event to be completed and then perform removing partitions.

**Pull request closed without merging**